nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/util/TlsHelper.java

```diff
@@ -0,0 +1,177 @@
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+
+public class TlsHelper {
+    public static final String PROVIDER = BouncyCastleProvider.PROVIDER_NAME;
+    public static final String PKCS12 = "PKCS12";
+    private final KeyPairGenerator keyPairGenerator;
+    private final int days;
+    private final String signingAlgorithm;
+
+    public TlsHelper(TlsHelperConfig tlsHelperConfig) throws NoSuchAlgorithmException {
+        this(tlsHelperConfig.getDays(), tlsHelperConfig.getKeySize(), tlsHelperConfig.getKeyPairAlgorithm(), tlsHelperConfig.getSigningAlgorithm());
+    }
+
+    public TlsHelper(TlsToolkitCommandLine tlsToolkitCommandLine) throws NoSuchAlgorithmException {
+        this(tlsToolkitCommandLine.getTlsHelperConfig());
+    }
+
+    public TlsHelper(int days, int keySize, String keyPairAlgorithm, String signingAlgorithm) throws NoSuchAlgorithmException {
+        this(createKeyPairGenerator(keyPairAlgorithm, keySize), days, signingAlgorithm);
+    }
+
+    protected TlsHelper(KeyPairGenerator keyPairGenerator, int days, String signingAlgorithm) {
+        this.keyPairGenerator = keyPairGenerator;
+        this.days = days;
+        this.signingAlgorithm = signingAlgorithm;
+    }
+
+    public static void addBouncyCastleProvider() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    private static KeyPairGenerator createKeyPairGenerator(String algorithm, int keySize) throws NoSuchAlgorithmException {
+        KeyPairGenerator instance = KeyPairGenerator.getInstance(algorithm);
+        instance.initialize(keySize);
+        return instance;
+    }
+
+    public KeyPair generateKeyPair() throws NoSuchAlgorithmException {
+        return keyPairGenerator.generateKeyPair();
+    }
+
+    public void addToKeyStore(KeyStore keyStore, KeyPair keyPair, String alias, char[] passphrase, Certificate... certificates) throws GeneralSecurityException, IOException {
+        keyStore.setKeyEntry(alias, keyPair.getPrivate(), passphrase, certificates);
+    }
+
+    public KeyStore createKeyStore(String keyStoreType) throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException, NoSuchProviderException {
+        KeyStore keyStore;
+        if (PKCS12.equals(keyStoreType)) {
+            keyStore = KeyStore.getInstance(keyStoreType, BouncyCastleProvider.PROVIDER_NAME);
+        } else {
+            keyStore = KeyStore.getInstance(keyStoreType);
+        }
+        keyStore.load(null, null);
+        return keyStore;
+    }
+
+    public X509Certificate generateSelfSignedX509Certificate(KeyPair keyPair, String dn) throws CertificateException {
+        return CertificateUtils.generateSelfSignedX509Certificate(keyPair, dn, signingAlgorithm, days);
+    }
+
+    public X509Certificate generateIssuedCertificate(String dn, PublicKey publicKey, X509Certificate issuer, KeyPair issuerKeyPair) throws CertificateException {
```

Comment:
I think in the instance methods `signingAlgorithm` and `days` should be passed as parameters so these methods can be `static`. Personally, I get nervous when these kinds of methods use hidden state.